pinot-core/src/main/java/com/linkedin/pinot/core/query/selection/SelectionFetcher.java

```diff
@@ -57,7 +57,7 @@
   private final SelectionColumnIterator[] selectionColumnIterators;
 
   public SelectionFetcher(Block[] blocks, DataSchema dataSchema) {
-    this.length = blocks.length;
+    this.length = dataSchema.size();
```

Comment:
Yes, this should not be changed. From the logic we assume  blocks.length == dataschema.size(). I updated the code and you can take a look at it.